src/components/catalog-search/catalog-search.controller.ts

```diff
@@ -0,0 +1,59 @@
+import * as _ from 'lodash';
+
+export class CatalogSearchController implements angular.IController {
+  static $inject = ['Catalog', 'KeywordService'];
+
+  public ctrl: any = this;
+
+  private Catalog: any;
+  private KeywordService: any;
+
+  constructor(Catalog: any, KeywordService: any) {
+    this.Catalog = Catalog;
+    this.KeywordService = KeywordService;
+  }
+
+  public $onInit() {
+    this.ctrl.searchText = '';
+  }
+
+  public $onChanges(onChangesObj: angular.IOnChangesObject) {
+    if (onChangesObj.serviceClasses || onChangesObj.imageStreams) {
+      // Convert service classes to ServiceItem, which is needed for the ordering panel.
+      let items = _.map(this.ctrl.serviceClasses, (serviceClass) => {
+        return this.Catalog.getServiceItem(serviceClass);
+      });
+
+      // Convert builders to ImageItem.
+      items = items.concat(_.map(this.ctrl.imageStreams, (imageStream) => {
+        return this.Catalog.getImageItem(imageStream);
+      }));
+
+      // Remove null items (non-builder images).
+      items = _.reject(items, (item) => {
+        return !item;
+      });
+
+      this.ctrl.allItems = _.sortBy(items, 'name');
```

Comment:
is it possible for someone to start typing into the filter while the service class or image streams are still loading? do you need to run filter here